instructure/canvas-android
# [Parent][MBL-14007] Add analytics

Also merged Logger class into analytics to be more concise. A lot of test consolidation too.

To test:
You can set your device as a debug device in firebase so you can see analytics coming in faster. It took me a couple tries to get it, but I followed the steps here: https://firebase.google.com/docs/analytics/debugview
You should see screen view events (the firebase_screen param is where our route name is shown)
Also should contain the login event, logout/switch users, adding student, help on the login screen, and refresh (not expiring yet, so won't happen).
